src/librustc_resolve/lib.rs

```diff
@@ -191,10 +189,6 @@ fn resolve_struct_error<'b, 'a: 'b, 'c>(resolver: &'b Resolver<'a>,
                                         span: syntax_pos::Span,
                                         resolution_error: ResolutionError<'c>)
                                         -> DiagnosticBuilder<'a> {
-    if !resolver.emit_errors {
-        return resolver.session.diagnostic().struct_dummy();
```

Comment:
@petrochenkov You can still do speculative resolution by passing `record_used: None` to `resolve_path`, `resolve_ident_in_lexical_scope`, or `resolve_name_in_module`.